server/src/main/java/org/opensearch/index/translog/InternalTranslogManager.java

```diff
@@ -430,10 +430,11 @@ public String getTranslogUUID() {
      * @return if the translog should be flushed
      */
     public boolean shouldPeriodicallyFlush(long localCheckpointOfLastCommit, long flushThreshold) {
-        final long translogGenerationOfLastCommit = translog.getMinGenerationForSeqNo(
-            localCheckpointOfLastCommit + 1
-        ).translogFileGeneration;
-        if (translog.sizeInBytesByMinGen(translogGenerationOfLastCommit) < flushThreshold) {
+        long minRefSeqNo = translog instanceof RemoteFsTranslog
+            ? ((RemoteFsTranslog) translog).getMinSeqNoToKeep()
```

Comment:
Can this logic move to `Translog` so that you get polymorphic behavior as opposed to doing the instanceof/casting logic here?

Reply:
@andrross In docrep, the min seq no to keep (as translog) consider is obtained using the segment infos which resides in engine. In case of remote store, the min seq no to keep stays in RemoteFsTranslog. Due to this, we had to handle this way. Let me know if you have any suggestion and I can incorporate it.

Reply:
Made the change as suggested. Please take a look, @Bukhtawar.